contracts/vesting/src/storage.rs

```diff
@@ -1,16 +1,121 @@
 use crate::errors::ContractError;
 use crate::vesting::Account;
-use cosmwasm_std::{Addr, Api, Storage};
+use crate::vesting::PledgeData;
+use cosmwasm_std::Order;
+use cosmwasm_std::{Addr, Api, Storage, Uint128};
 use cw_storage_plus::{Item, Map};
 
 pub const KEY: Item<u32> = Item::new("key");
 const ACCOUNTS: Map<Addr, Account> = Map::new("acc");
+// Holds data related to individual accounts
+const BALANCES: Map<u32, Uint128> = Map::new("blc");
+const BOND_PLEDGES: Map<u32, PledgeData> = Map::new("bnd");
+const GATEWAY_PLEDGFES: Map<u32, PledgeData> = Map::new("gtw");
+const DELEGATIONS: Map<(u32, &[u8], u64), Uint128> = Map::new("dlg");
```

Comment:
1. Could we use some type aliases for the types used in the key? I had to go looking for what `&[u8]` and `u64` are supposed to represent here.
2. Isn't it possible to instead of `&[u8]` use the `IdentityKey` (`String` alias) type?
3. Is block `time` what we want for the third part of the key? Maybe `height` is more appropriate? (note:  I have absolutely no idea, it just felt weird to be that we are looking up values by time, which isn't super predictable). Should it even be part of the key to begin with? Perhaps it would be better as a value?
4. Do we even need a composite key given that the storage key, generated via `generate_storage_key` is bound to be unique for each delegation?

Reply:
1. Done,
2. Yup
3. `height` is probably a better choice
4. Its possible for them to delegate to the same node more then once, so we keep track of each individual delegation due to vesting accounting, ie we want to be able to say how many delegated coins are vested vs unvested, although the way that is implemented could be reworked to just store delegation total.

Reply:
I'll resolve this and merge, but lets continue the discussion if we need to, I just don't want this sitting in here for too long. Current storage is better then it was at least memory wise, we can continue to optimise it